cpp/src/address.cpp

```diff
@@ -38,21 +38,21 @@ std::shared_ptr<Address> createAddressFromWorker(std::shared_ptr<Worker> worker)
   return std::shared_ptr<Address>(new Address(worker, address, length));
 }
 
-std::shared_ptr<Address> createAddressFromString(std::string addressString)
+std::shared_ptr<Address> createAddressFromString(std::string_view addressString)
 {
-  ucp_address_t* address = reinterpret_cast<ucp_address_t*>(new char[addressString.length()]);
   size_t length          = addressString.length();
-  memcpy(address, addressString.c_str(), length);
+  ucp_address_t* address = reinterpret_cast<ucp_address_t*>(new char[length]);
+  memcpy(address, addressString.data(), length);
   return std::shared_ptr<Address>(new Address(nullptr, address, length));
 }
```

Comment:
This is fine.

Reply:
For additional context, this is used in one place in Cython where we already have something that could be `std::string_view` compliant (a Python `bytes` object). Currently in the construction we...

1. Copy the `bytes` object into a `std::string`
2. Copy the `std::string` to the `createAddressFromString` function call (no `const std::string&` currently)
3. Allocate and copy the `std::string` to a `char[]`

https://github.com/rapidsai/ucxx/blob/4cd6354fc23e62f19b9715c04fd69dd1a1298997/python/ucxx/ucxx/_lib/libucxx.pyx#L407-L416

4. Note there is a `getString` call at the end doing one more copy